### PR TITLE
Build the net6 templates in net7

### PR DESCRIPTION
**Pull request by mattleibow**
### Description of Change

Re-enable building net6.0-android, net6.0-ios and net6.0-maccatalyst in the net7.0 pipeline.

### CODE REVIEW

**Review by TanayParikh — APPROVED** [review]

File: eng/cake/dotnet.cake [hunk 1]
```diff
@@ -188,21 +188,6 @@ Task("dotnet-templates")
...
-                    // Mac Catalyst/iOS does not yet support net6 from the net7 SDK.
-                    ReplaceTextInFiles($"{projectName}/*.csproj",
-                        ";net6.0-ios;net6.0-maccatalyst",
-                        "");
-
-                    // Android does not yet support net6 from the net7 SDK in Release builds
-                    if (configuration == "Release")
-                        ReplaceTextInFiles($"{projectName}/*.csproj",
-                            "net6.0-android",
-                            "");
```
Comment (jonathanpeppers) [inline eng/cake/dotnet.cake:203]: You will probably have to put this `if` back in, this problem is caused by new linker behavior in .NET 7, but this isn't backported to 6:

https://github.com/dotnet/runtime/pull/72143#issuecomment-1233715839

**Comment by rmarinho** [comment]
Do we still need this one @mattleibow  ?

**Comment by mattleibow** [comment]
Ah yeah. I need to rebase and see if all the build combos work. I know we had to disable a bunch for rc1, but should all be fixed in rc2.

File: src/Controls/src/Core/BindableProperty.cs [hunk 2]
```diff
@@ -41,7 +41,6 @@ public sealed class BindableProperty
 		static readonly Dictionary<Type, TypeConverter> KnownTypeConverters = new Dictionary<Type, TypeConverter>
 		{
 			{ typeof(Uri), new UriTypeConverter() },
-			{ typeof(Easing), new Maui.Converters.EasingTypeConverter() },
```
Comment (rmarinho) [inline src/Controls/src/Core/BindableProperty.cs:44]: Why this remove?

**Review by rmarinho — APPROVED** [review]

**Pull request merged**